allocation.md

```diff
@@ -1,53 +1,406 @@
+```
+http://<kubecost>/model/allocation
+```
+
+> **NOTE**
+>
+> Throughout, we use `localhost:9090` as the default Kubecost URL, but your Kubecost instance may be exposed by a service or ingress. To reach Kubecost at port 9090, run: `kubectl port-forward deployment/kubecost-cost-analyzer -n kubecost 9090`
+
+## Quick start
+
+Request allocation data for each 24-hour period in the last three days, aggregated by namespace:
+```
+$ curl http://localhost:9090/model/allocation \
+  -d window=3d \
+  -d aggregate=namespace \
+  -d accumulate=false \
+  -d shareIdle=false
+  -G
+```
+```json
+{
+  "code": 200,
+  "data": [
+    {
+      "__idle__": { ... },
+      "default": { ... },
+      "kube-system": { ... },
+      "kubecost": { ... }
+    },
+    {
+      "__idle__": { ... },
+      "default": { ... },
+      "kube-system": { ... },
+      "kubecost": { ... }
+    },
+    {
+      "__idle__": { ... },
+      "default": { ... },
+      "kube-system": { ... },
+      "kubecost": { ... }
+    },
+    {
+      "__idle__": { ... },
+      "default": { ... },
+      "kube-system": { ... },
+      "kubecost": { ... }
+    }
+  ]
+}
+```
+
+Note: querying for "3d" will likely return a range of four sets because the queried range will overlap with four precomputed 24-hour sets, each aligned to the configured timezone. For instance, querying "3d" on 2021/01/04T12:00:00 will return:
+- 2021/01/04 00:00:00 until 2021/01/04T12:00:00 (now)
+- 2021/01/03 00:00:00 until 2021/01/04 00:00:00
+- 2021/01/02 00:00:00 until 2021/01/03 00:00:00
+- 2021/01/01 00:00:00 until 2021/01/02 00:00:00
+
+See [Querying](#querying) for the full list of arguments and [Examples](#examples) for more example queries.
+
+## Allocation schema (version 1.76)
+Field | Description
+---: | :---
+name | Name of each relevant Kubernetes concept described by the allocation, delimited by slashes, e.g. "cluster/node/namespace/pod/container"
+properties | Map of name-to-value for all relevant property fields, including: `cluster`, `node`, `namespace`, `controller`, `controllerKind`, `pod`, `container`, `labels`, `annotation`, etc.
+window | Period of time over which the allocation is defined.
+start | Precise starting time of the allocation. By definition must be within the window.
+end | Precise ending time of the allocation. By definition must be within the window.
+minutes | Number of minutes running; i.e. the minutes from `start` until `end`.
+cpuCores | Average number of CPU cores allocated while running.
+cpuCoreRequestAverage | Average number of CPU cores requested while running.
+cpuCoreUsageAverage | Average number of CPU cores used while running.
+cpuCoreHours | Cumulative CPU core-hours allocated.
+cpuCost | Cumulative cost of allocated CPU core-hours.
+cpuEfficiency | Fraction of CPU request that was used.
```

Comment:
Is this "Average fraction of CPU request that was used while running"?

Reply:
If we want to use the word "average", it would be most accurate to say: "Ratio of `cpuCoreUsageAverage`-to-`cpuCoreRequestAverage`, meant to represent the fraction of requested resources that were used." Is that better, or too wordy?

Reply:
I definitely prefer that rewording.
```suggestion
cpuEfficiency | Ratio of `cpuCoreUsageAverage`-to-`cpuCoreRequestAverage`, meant to represent the fraction of requested resources that were used.
```